example/provider-extensions/garden/configure-admission.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+#
+# Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses~LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+REPO_ROOT_DIR="$(realpath "$SCRIPT_DIR"/../../..)"
+
+usage() {
+  echo "Usage:"
+  echo "> configure-admission.sh [ -h | <garden-kubeconfig> <apply|delete> ]"
+  echo
+  echo ">> For example: configure-admission.sh ~/.kube/garden-kubeconfig.yaml apply"
+
+  exit 0
+}
+
+if [ "$1" == "-h" ] || [ "$#" -ne 2 ]; then
+  usage
+fi
+
+if [ "$2" != "apply" ] && [ "$2" != "delete" ]; then
+  usage
+fi
+
+garden_kubeconfig=$1
+command=$2
+
+for p in $(yq '. | select(.kind == "ControllerDeployment") | select(.metadata.name == "provider-*") | .metadata.name' $REPO_ROOT_DIR/example/provider-extensions/garden/controllerregistrations/* | grep -v -E "^---$"); do
```

Comment:
```suggestion
for p in $(yq '. | select(.kind == "ControllerDeployment") | select(.metadata.name == "provider-*") | .metadata.name' "$REPO_ROOT_DIR"/example/provider-extensions/garden/controllerregistrations/* | grep -v -E "^---$"); do
```
Same for other variables in paths. Otherwise, it fails in case someone has spaces in their paths.

Reply:
Fair point. Adapted with https://github.com/gardener/gardener/pull/8311/commits/89402c058b31edd6af49bd0adbce94d7d2e22a3b.